# Titanic Data Set Part 1
### Neel Shettigar
### 04/18/2021

This program answers questions related to the Titanic Data Set.

In [7]:
# Import libraries 
import pandas as pd

# Read data into DataFrame
df = pd.read_csv('titanic.csv')

# Create a subset of children 10 and under in third class
subset = df.loc[(df['Pclass'] == 3) & (df['Age'] <= 10)]
# Count of the children who survived and died
survived = len(subset[subset.Survived == 1]) 
died = len(subset[subset.Survived == 0])

# Average fare of everyone on the ship
cost = df['Fare'].mean()

# Expectation/Average fare conditioned on passenger-class(lowest paid = 3rd class, Middle paid = 2nd Class, Highest paid = 1st class)
lowest = (df.loc[df['Pclass'] == 3])['Fare'].mean()
middle = (df.loc[df['Pclass'] == 2])['Fare'].mean()
highest = (df.loc[df['Pclass'] == 1])['Fare'].mean()

# Display calculated values
print("Probability: " + str(round((((survived / (survived + died)) * 100)),2)) + "%")
print("Average Ship Fare Cost: £" + str(round(cost,2)))
print("Average Third Class Fare Cost: £" + str(round(lowest,2)))
print("Average Second Class Fare Cost: £" + str(round(middle,2)))
print("Average First Class Fare Cost: £" + str(round(highest,2)))


Probability: 41.51%
Average Ship Fare Cost: £32.31
Average Third Class Fare Cost: £13.71
Average Second Class Fare Cost: £20.66
Average First Class Fare Cost: £84.15


## 1. What is the probability that a child who is in third class and is 10 years old or younger survives?
#### The probability that a child who is in third class and is 10 years old or younger is around 41.51%. This was calculated by dividing the total number of survivors by the total number of casualties and then multiplying by 100. 

## 2. How much did people pay to be on the ship (average)? 
#### The people paid an average(mean) of around £32.31 to be on the ship. This was calculated by taking the average of all the rows in the 'Fare' column.

## 3. Calculate the expectation of fare conditioned on passenger-class (lowest, middle and highest paid).
#### The expected fare(mean) of 3rd class(lowest paid) was around £13.71, the expected fare(mean) of 2nd class(middle paid) was around £20.66 and the expected fare(mean) of 1st class(highest paid) was around £84.15. This was calculated by conditioning the rows by passenger-class, then finding the average of their respective 'Fare' columns.